drag_and_drop_v2/utils.py

```diff
@@ -5,3 +5,135 @@
 # Make '_' a no-op so we can scrape strings
 def _(text):
     return text
+
+
+class Constants(object):
+    ALLOWED_ZONE_ALIGNMENT = ['left', 'right', 'center']
+    DEFAULT_ZONE_ALIGNMENT = 'center'
+
+
+class StateMigration(object):
+    """
+    Helper class to apply zone data and item state migrations
+    """
+
+    @classmethod
+    def _apply_migration(cls, obj, migrations):
+        """
+        Applies migrations sequentially to a copy of an `obj`, to avoid updating actual data
+        """
+        tmp = obj.copy()
+        for method in migrations:
+            tmp = method(tmp)
+
+        return tmp
+
+    @classmethod
+    def apply_zone_migrations(cls, zone):
+        """
+        Applies zone migrations
+        """
+        migrations = (cls._zone_v1_to_v2, cls._zone_v2_to_v2p1)
+
+        return cls._apply_migration(zone, migrations)
+
+    @classmethod
+    def apply_item_state_migrations(cls, item_state):
+        """
+        Applies item_state migrations
+        """
+        migrations = (cls._item_state_v1_to_v1p5, cls._item_state_v1p5_to_v2, cls._item_state_v1p5_to_v2p1)
+
+        return cls._apply_migration(item_state, migrations)
+
+    @classmethod
+    def _zone_v1_to_v2(cls, zone):
+        """
+        Migrates zone data from v1.0 format to v2.0 format.
+
+        Changes:
+        * v1 used zone "title" as UID, while v2 have "uid" property
+        * "id" and "index" properties are no longer used
+
+        In: {'id': 1, 'index': 2, 'title': "Zone", ...}
+        Out: {'uid': "Zone", ...}
+        """
+        if "uid" not in zone:
+            zone["uid"] = zone.get("title")
+        zone.pop("id", None)
+        zone.pop("index", None)
+
+        return zone
+
+    @classmethod
+    def _zone_v2_to_v2p1(cls, zone):
+        """
+        Migrates zone data from v2.0 to v2.5
```

Comment:
@e-kolpakov I think this should be `v2.0` to `v.2.1`. Same comment applies to the docstring of `_item_state_v1p5_to_v2p1` below.